indico/modules/events/registration/util.py

```diff
@@ -422,3 +425,185 @@ def build_registration_api_data(registration):
     registration_info['checkin_date'] = registration.checked_in_dt.isoformat() if registration.checked_in_dt else ''
     registration_info['event_id'] = registration.event_id
     return registration_info
+
+
+class RegistrationListGenerator(ListGeneratorBase):
+    """Listing and filtering actions in the registration list."""
+
+    endpoint = '.manage_reglist'
+    list_link_type = 'registration'
+
+    def __init__(self, event, regform):
+        super(RegistrationListGenerator, self).__init__(event, entry_parent=regform)
+        self.regform = regform
+        self.default_list_config = {
+            'items': ('title', 'email', 'affiliation', 'reg_date', 'state'),
+            'filters': {'fields': {}, 'items': {}}
+        }
+        self.special_items_info = OrderedDict([
+            ('reg_date', {
+                'title': _('Registation Date'),
+                'id': 'reg_date'
+            }),
+            ('price', {
+                'title': _('Price'),
+                'id': 'price'
+            }),
+            ('state', {
+                'title': _('State'),
+                'id': 'state',
+                'filter_choices': {str(state.value): state.title for state in RegistrationState}
+            }),
+            ('checked_in', {
+                'title': _('Checked in'),
+                'id': 'checked_in',
+                'filter_choices': {
+                    '0': _('No'),
+                    '1': _('Yes')
+                }
+            }),
+            ('checked_in_date', {
+                'title': _('Check-in date'),
+                'id': 'checked_in_date'
+            })
+        ])
+        self.basic_items = ('title', 'first_name', 'last_name', 'email', 'position', 'affiliation', 'address', 'phone',
+                            'country')
+        self.special_items = tuple(self.special_items_info)
+        self.list_config = self._get_config()
+
+    def _get_non_custom_columns(self, ids):
```

Comment:
yeah well... It's a very complicated situation.. We have the `custom` columns which come from the custom registration fields, the `special` columns which are info like checkin, registration state, date, etc., and `basic` which are personal info about the registrant. The above function returns the columns that are not custom so basically the `special` and `basic`